From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: OpenWrt <openwrt@openwrt.org>
Date: Wed, 25 Jun 2026 00:00:00 +0000
Subject: arm64: dts: rockchip: CM5 Base FPC I2C pull-ups + OLED RST pinctrl

The CM5 Base 12-pin FPC has no external I2C pull-ups; enable SoC
bias-pull-up on GPIO4_B2/B3 so i2cdetect shows "--" on an idle bus.
Also mux GPIO4_B4 for waveshare-oled-rst (patch 999).

---
--- a/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-cm5-base.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-cm5-base.dts
@@ -339,6 +339,20 @@
 		vbus_5v0_en_pin: vbus-5v0-en-pin {
 			rockchip,pins = <0 RK_PD3 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	fpc {
+		fpc_i2c7_xfer: fpc-i2c7-xfer {
+			rockchip,pins =
+				<4 RK_PB2 9 &pcfg_pull_up_smt>,
+				<4 RK_PB3 9 &pcfg_pull_up_smt>;
+		};
+	};
+
+	oled {
+		waveshare_oled_rst_pin: waveshare-oled-rst-pin {
+			rockchip,pins = <4 RK_PB4 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
 };
